server/src/main/java/org/opensearch/index/translog/Translog.java

```diff
@@ -650,6 +650,14 @@ public Snapshot newSnapshot(long fromSeqNo, long toSeqNo, boolean requiredFullRa
                 .filter(reader -> reader.getCheckpoint().minSeqNo <= toSeqNo && fromSeqNo <= reader.getCheckpoint().maxEffectiveSeqNo())
                 .map(BaseTranslogReader::newSnapshot)
                 .toArray(TranslogSnapshot[]::new);
+            if (this instanceof RemoteFsTranslog
+                && fromSeqNo != 0
+                && Arrays.stream(snapshots)
+                    .noneMatch(
+                        snapshot -> snapshot.getCheckpoint().minSeqNo <= fromSeqNo || snapshot.getCheckpoint().maxSeqNo >= fromSeqNo
+                    )) {
+                logger.warn("No operation from from_seqno {} found", fromSeqNo);
```

Comment:
Missing operations from start_seq_no to end_seq_no?

Reply:
No. We read SegmentInfos file and fetch sequence number from user data and pass it as from_seqno. Here, we check if the from_seqno is present in translog. This will only detect hole at the start of the translog.